pkg/discovery/worker/metrics_collector.go

```diff
@@ -0,0 +1,343 @@
+		if pod != nil {
+			podNames = podNames + "," + pod.PodName
+		}
+	}
+	return podNames
+}
+
+// Returns the sum of allocation resources for all the pods in the collection
+func (podMetricsList PodMetricsList) SumAllocationUsage() map[metrics.ResourceType]float64 {
+	allocationResourcesSum := make(map[metrics.ResourceType]float64)
+
+	// first sum the compute resources used
+	computeResourcesSum := make(map[metrics.ResourceType]float64)
+	for _, computeType := range metrics.ComputeResources {
+		var totalUsed float64
+		for _, podMetrics := range podMetricsList {
+			if podMetrics.ComputeUsed == nil {
+				continue
+			}
+			// check if allocation bought and the resource type exists
+			used, exists := podMetrics.ComputeUsed[computeType]
+			if !exists {
+				continue
+			}
+			totalUsed += used
+		}
+		computeResourcesSum[computeType] = totalUsed
+	}
+
+	// pod allocation resource usage is same as its corresponding compute resource usage
+	for _, allocationType := range metrics.ComputeAllocationResources {
+		computeType, exists := metrics.AllocationToComputeMap[allocationType]
+		if !exists {
+			fmt.Errorf("cannot find corresponding compute type for %s", allocationType)
+			continue
+		}
+		allocationResourcesSum[allocationType] = computeResourcesSum[computeType]
+	}
+
+	glog.V(4).Infof("Collected allocation resources for pod collection %s\n", podMetricsList.GetPods())
+	for rt, used := range allocationResourcesSum {
+		glog.V(4).Infof("\t rt=%s used=%f\n", rt, used)
+	}
+	return allocationResourcesSum
+}
+
+func (podCollectionMap PodMetricsByNodeAndQuota) addPodMetric(podName, nodeName, quotaName string,
+								podMetrics *repository.PodMetrics) {
+	podsByQuotaMap, exists := podCollectionMap[nodeName]
+	if !exists {
+		// create the map for this node
+		podCollectionMap[nodeName] = make(map[string]PodMetricsList)
+	}
+	podsByQuotaMap, _ = podCollectionMap[nodeName]
+
+	_, exists = podsByQuotaMap[quotaName]
+	if !exists {
+		// create the map for this quota
+		podsByQuotaMap[quotaName] = PodMetricsList{}
+	}
+	podMetricsList, _ := podsByQuotaMap[quotaName]
+	podMetricsList = append(podMetricsList, podMetrics)
+	podsByQuotaMap[quotaName] = podMetricsList
+	podCollectionMap[nodeName] = podsByQuotaMap
+	glog.V(4).Infof("Created pod metrics for %s, quota=%s, node=%s\n", podName, quotaName, nodeName)
+}
+
+// -------------------------------------------------------------------------------------------------
+
+// Create Pod metrics by selecting the pod compute resource usages from the metrics sink.
+// The PodMetrics are organized in a map by node and quota
+func (collector *MetricsCollector) CollectPodMetrics() PodMetricsByNodeAndQuota {
+	podCollectionMap := make(PodMetricsByNodeAndQuota)
+	// Iterate over all pods
+	for _, pod := range collector.PodList {
+		// Find quota entity for the pod if available
+		quota := collector.Cluster.GetQuota(pod.ObjectMeta.Namespace)
+		if quota == nil {
+			//ignore the pod not associated with a quota
```

Comment:
This shouldn't happen, correct? Or this can happen under some condition?

Reply:
added error message